pkg/composableschemadsl/compiler/translator.go

```diff
@@ -20,14 +21,13 @@ import (
 )
 
 type translationContext struct {
-	objectTypePrefix     *string
-	mapper               input.PositionMapper
-	schemaString         string
-	skipValidate         bool
-	existingNames        *mapz.Set[string]
-	locallyVisitedFiles  *mapz.Set[string]
-	globallyVisitedFiles *mapz.Set[string]
-	sourceFolder         string
+	objectTypePrefix    *string
+	mapper              input.PositionMapper
+	schemaString        string
+	skipValidate        bool
+	existingNames       *mapz.Set[string]
+	compiledPartials    map[string][]*core.Relation
+	partialWaitingRooms *mapz.MultiMap[string, *dslNode]
```

Comment:
These are for the partial compilation process. `partialWaitingRooms` could probably move into the `translatePartials` function if desired.

Reply:
why is it called a waiting room?

Reply:
It was a name i reached for. The idea is that the values of a given key are the partials whose compilation depends on the compilation of the partial named in the key, and my brain went to "waiting room" for that concept. I can change that to "unresolvedPartials" or something if that's preferable.